uber/site_sections/hotel_reports.py

```diff
@@ -529,6 +529,16 @@ def requested_hotel_info(self, out, session):
             headers.append('LoginID{}'.format(count))
 
         out.writerow(headers)
-        added = []
+        added = set()
         for a in sorted(hotel_query.all(), key=lambda a: a.legal_name or a.full_name):
-            out.writerow([a.legal_first_name, a.legal_last_name, a.email, a.hotel_pin])
+            row = [a.legal_first_name, a.legal_last_name, a.email, a.hotel_pin]
+            if a.hotel_pin not in added:
+                added.add(a.hotel_pin)
+                for matching_attendee in \
+                        hotel_query.filter(Attendee.email == a.email,
+                                           Attendee.id != a.id,
+                                           Attendee.last_name == a.last_name):
```

Comment:
You'll notice that we don't check for matching_attendees' first names to be the same as the attendee we're checking against in this filter. That's because, as far as I can tell, _that's the performance problem_. Seriously, if I add the line `Attendee.first_name == a.first_name`, the export will 504, and if I don't, it will complete in under 30 seconds.

Yea, I don't know either.